src/components/Visualization/useAnalyticsData.js

```diff
@@ -43,8 +43,21 @@ const excludedDimensions = [
     DIMENSION_ID_LAST_UPDATED_BY,
 ]
 
-const findOptionSetItem = (code, metaDataItems) =>
-    Object.values(metaDataItems).find((item) => item.code === code)
+const findOptionSetItem = (optionSetId, code, metaDataItems) => {
+    const optionSetMetaData = metaDataItems?.[optionSetId]
+
+    if (optionSetMetaData) {
+        const optionId = optionSetMetaData.options.find(
+            (option) => option.code === code
+        )?.uid
+
+        if (optionId) {
+            return metaDataItems[optionId]
+        }
+    }
+
+    return undefined
```

Comment:
Given how this fn is used I don't think you need to return anything here?
But if you want to make it explicit, I think the correct return value would be `null`.

Reply:
My reasoning was to return the same value the function can return above where the `find` is used.